editoast/src/views/rolling_stocks.rs

```diff
@@ -84,28 +90,91 @@ impl From<RollingStockCreateForm> for RollingStockModel {
     }
 }
 
+impl RollingStockForm {
+    fn into_rolling_stock_model(self, rolling_stock_id: i64) -> RollingStockModel {
+        RollingStockModel {
+            id: Some(rolling_stock_id),
+            name: Some(self.name),
+            version: Some(self.version),
+            effort_curves: Some(DieselJson(self.effort_curves)),
+            base_power_class: Some(self.base_power_class),
+            length: Some(self.length),
+            max_speed: Some(self.max_speed),
+            startup_time: Some(self.startup_time),
+            startup_acceleration: Some(self.startup_acceleration),
+            comfort_acceleration: Some(self.comfort_acceleration),
+            gamma: Some(DieselJson(self.gamma)),
+            inertia_coefficient: Some(self.inertia_coefficient),
+            features: Some(self.features),
+            mass: Some(self.mass),
+            rolling_resistance: Some(DieselJson(self.rolling_resistance)),
+            loading_gauge: Some(self.loading_gauge),
+            metadata: Some(DieselJson(self.metadata)),
+            power_restrictions: Some(self.power_restrictions),
+            ..Default::default()
+        }
+    }
+}
+
+fn check_effort_curves(effort_curves: EffortCurves) -> Result<()> {
+    for (_, mode_effort_curves) in effort_curves.modes {
+        let default_curve = mode_effort_curves.default_curve;
+        if default_curve.max_efforts.len() != default_curve.speeds.len() {
+            return Err(RollingStockError::InvalidEffortCurve.into());
+        }
+        for curve in mode_effort_curves.curves {
+            if curve.curve.max_efforts.len() != curve.curve.speeds.len() {
+                return Err(RollingStockError::InvalidEffortCurve.into());
+            }
+        }
+    }
+    Ok(())
+}
```

Comment:
Maybe this function should be in src/schema, to discuss

Reply:
Ideally, we would have a custom deserialization that would check this.
Otherwise, we can move this function to `src/schema` and make it an implementation of `EffortCurves`.